### ISSUE TITLE
Assignment to ternary expression using function parameter as condition triggers no-param-reassign

### ISSUE THREAD

**Issue by zhouhaoyu** [eslint/eslint#11236]
**Tell us about your environment**

* **ESLint Version:** 5.9.0
* **Node Version:** 8.12.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parser": "babel-eslint",
    "extends": "airbnb",
    "env": {
        "browser": true
    },
    "rules": {
        "jsx-a11y/click-events-have-key-events": "off",
        "jsx-a11y/no-static-element-interactions": "off",
        "react/jsx-one-expression-per-line": [true, { "allow": "literal" }],
        "no-restricted-syntax": ["error", "ForInStatement", "LabeledStatement", "WithStatement"]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function test(arg) {
  const array = [[1], [2]];
  (arg ? array[0] : array[1])[0] = 3;
  return array;
}
```

```bash
yarn eslint src/
```

**What did you expect to happen?**
The above code should not emit lint errors.

**What actually happened? Please include the actual, raw output from ESLint.**
`error  Assignment to property of function parameter 'arg'     no-param-reassign`

**Are you willing to submit a pull request to fix this bug?**
Will try if possible.

**Comment by not-an-aardvark** [comment]
Thanks for the report, I can reproduce this issue. It seems to only occur when using the `props: true` option.

**Comment by platinumazure** [comment]
Hi @zhouhaoyu, thanks for the issue!

I [can reproduce](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXBhcmFtLXJlYXNzaWduOiBbZXJyb3IsIHsgcHJvcHM6IHRydWUgfV0gKi9cblxuZnVuY3Rpb24gdGVzdChhcmcpIHtcbiAgY29uc3QgYXJyYXkgPSBbWzFdLCBbMl1dO1xuICAoYXJnID8gYXJyYXlbMF0gOiBhcnJheVsxXSlbMF0gPSAzO1xuICByZXR1cm4gYXJyYXk7XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19) in the demo, so I'm marking this as accepted. Thanks also for the PR-- I'll review that shortly!
